packages/plugin-dominos/src/actions/startOrder.ts

```diff
@@ -0,0 +1,275 @@
+import {
+    Action,
+    ActionExample,
+    composeContext,
+    generateObjectV2,
+    Handler,
+    IAgentRuntime,
+    Memory,
+    ModelClass,
+    State,
+} from "@ai16z/eliza";
+import { Customer, Item, Order } from "dominos";
+import { PizzaCrust, PizzaSize } from "../types";
+
+import { z } from "zod";
+import { PizzaOrderManager } from "../PizzaOrderManager";
+
+const handler: Handler = async (
+    runtime: IAgentRuntime,
+    message: Memory,
+    state: State
+) => {
+    const orderManager = new PizzaOrderManager(runtime);
+    const userId = message.userId;
+
+    // Check for existing order
+    const existingOrder = await orderManager.getOrder(userId);
+    if (existingOrder) {
+        return "There is already an active order. Please complete or cancel the existing order before starting a new one.";
+    }
+
+    console.log("Existing order: ", existingOrder);
+
+    // Extract order details from message using LLM
+    const extractionTemplate = `
```

Comment:
Would consider moving the template and the schema definition to a new file template to templates.ts and schema to types.ts